projects/kit/components/badge/badge.component.ts

```diff
@@ -72,4 +72,8 @@ export class TuiBadgeComponent {
     get isEmpty(): boolean {
         return this.value === ``;
     }
+
+    titleText({offsetWidth, scrollWidth}: HTMLElement): string {
```

Comment:
what do you think about?

```ts
@tuiPure
titleText(offsetWidth, scrollWidth): string {
  return offsetWidth < scrollWidth ? this.outputValue : ``;
}
```

Reply:
It doesn't recalculate the value with tuiPure (
but it is safe without tuiPure, it only executes on resize